File: selfstudy/js/players_filterhigh.js
/*
task 65:
Players - FilterHigh
You need to return only the values greater than 50 from player heights in cm
*/

function above_fifty_players(heights) {
  return heights.filter(x => x > 50);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: above_fifty_players([10, 20, 30]) },
  { input: [50, 60, 70], expected: above_fifty_players([50, 60, 70]) },
  { input: [1, 2, 3], expected: above_fifty_players([1, 2, 3]) },
  { input: [100, 200], expected: above_fifty_players([100, 200]) },
  { input: [], expected: above_fifty_players([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('above_fifty_players', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(above_fifty_players(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, above_fifty_players(t.input) === t.expected || JSON.stringify(above_fifty_players(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think

File: selfstudy/golang/players_filterhigh.go
/*
task 265:
Players - FilterHigh
You need to return only the values greater than 50 from player heights in cm
*/

package main

import (
	"fmt"
	"reflect"
)

func Above_fiftyPlayers(heights []int) []int {
	var res []int
	for _, v := range heights {
		if v > 50 { res = append(res, v) }
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Above_fiftyPlayers([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Above_fiftyPlayers([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Above_fiftyPlayers([]int{1, 2, 3})},
		{[]int{100, 200}, Above_fiftyPlayers([]int{100, 200})},
		{[]int{}, Above_fiftyPlayers([]int{})},
	}
	
	for idx, t := range tests {
		res := Above_fiftyPlayers(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think
